source/gui.py

```diff
@@ -0,0 +1,332 @@
+                new_shape_shape = ''        # ILLEGAL
+                new_shape_pos   = loc.Pos() # ILLEGAL
+                new_shape_radius=10
+                new_shape_rot_rad=0.0
+                new_shape_depth =70
+
+                # Apply properties to shape
+                found = False
+                for shape in self.list_of_canvas_shapes: # Find a shape where we drop
+                    if event.x < shape.box.pos.x: continue                      # Left of box  (out of range)
+                    if event.x > shape.box.pos.x + shape.box.size.x: continue   # Right of box (out of range)
+                    if event.y < shape.box.pos.y: continue                      # Top of box   (out of range)
+                    if event.y > shape.box.pos.y + shape.box.size.y: continue   # Bottom of box(out of range)
+
+                    new_shape_shape = shapes.object_names_array[int(shape.class_id)].replace(' ', '_')
+                    if PalletItem.YELLOW.value <= pallet_item.value <= PalletItem.BLUE.value:           new_shape_color = pallet_item.name.lower()
+                    if PalletItem.CIRCLE.value <= pallet_item.value <= PalletItem.HALF_CIRCLE.value:    new_shape_shape = pallet_item.name.lower()
+
+                    shape.remove_shape(self)
+                    self.list_of_canvas_shapes.remove(shape)
+
+                    found = True
+                # Create a new shape
+                new_shape=None
+                if not found:
+                    if PalletItem.YELLOW.value <= item.value <= PalletItem.BLUE.value: # TODO: Change background
+                        self.in_hand.remove(item)
+                        continue
+                    # else place a new shape
+                    new_shape = get_new_shape(shape=item,
+                                              center_pos=loc.Pos(event.x, event.y),
+                                              size=None, color=None)
+                else:# So we did find it
+                    if PalletItem.YELLOW.value <= item.value <= PalletItem.BLUE.value: # Apply color to shape 
+                        new_shape = get_new_shape(shape=PalletItem(int(shape.class_id)+PalletItem.CIRCLE.value),
+                                                  center_pos=shape.center_pos,
+                                                  size=shape.box.size, color=item)
+                    else: # Apply shape to shape 
+                        new_shape = get_new_shape(shape=item,
+                                                  center_pos=shape.center_pos,
+                                                  size=shape.box.size, color=PalletItem[shape.fill_color.upper()])
+                new_shape.draw_shape(tkinter_canvas=self,
+                                     location_offset=loc.Pos(x=self.pallet_item_size().x,y=0))
+                self.list_of_canvas_shapes.append(new_shape)
+                self.in_hand.remove(item)
```

Comment:
Does a function in python always automatically return? without return at the end?